selectel/mks.go

```diff
@@ -238,25 +289,28 @@ func upgradeMKSClusterV1KubeVersion(ctx context.Context, d *schema.ResourceData,
 			return fmt.Errorf("error getting incremented minor part of the current version %s: %s", currentVersion, err)
 		}
 
-		// Get latest patch versions for every minor version.
-		latestPatchVersions, err := mksClusterV1GetLatestPatchVersions(ctx, client)
+		// Check that next minor version is equal to desire version.
+		if currentMinorNew != desiredMinor {
+			return fmt.Errorf("invalid minor version: %s, kubernets versions must be upgraded one by one", desiredMinor)
+		}
+
+		// Check that new minor version is supported.
+		isSupported, err := checkVersionIsSupported(ctx, client, desiredVersion)
 		if err != nil {
-			return fmt.Errorf("error getting latest patch versions: %s", err)
+			return fmt.Errorf("can't check support for version: %s", err)
 		}
 
-		// Check that we have a Kubernetes version of the current minor version + 1.
-		latestVersion, ok := latestPatchVersions[currentMinorNew]
-		if !ok {
-			return fmt.Errorf("the cluster is already on the latest available minor version: %s", currentMinor)
+		if !isSupported {
+			log.Print("[DEBUG] cluster will be upgrade to unsupported version. Patch version will selected automatically.")
 		}
 
-		log.Printf("[DEBUG] latest kube version: %s", latestVersion)
+		latestMinorVersion, err := mksClusterV1GetLatestMinorVersion(ctx, client)
+		if err != nil {
+			return fmt.Errorf("can't find latest minor version: %s", err)
+		}
 
-		// Compare the latest patch version with the desired version.
-		if desiredVersion != latestVersion {
-			return fmt.Errorf(
-				"current version %s can't be upgraded to version %s, the latest available version is: %s",
-				currentVersion, desiredVersion, latestVersion)
+		if latestMinorVersion == currentMinor {
+			return fmt.Errorf("the cluster is already on the latest available minor version: %s", currentMinor)
 		}
```

Comment:
It seems that we can do this check earlier at 283 line after this code:
```
if desiredMinor != currentMinor {
		log.Print("[DEBUG] upgrading minor version")
```

Reply:
moved